fpdf/output.py

```diff
@@ -434,6 +436,106 @@ def serialize(self, _security_handler=None):
         return "\n".join(out)
 
 
+class ICCProfileStreamDict():
+
+    __slots__ = (  # RAM usage optimization
+        "fn",
+        "N",
+        "alternate",
+    )
+
+    def __init__(self, fn=None, N=None, alternate=None):
+        """hold values from input for dest_output_profile
+
+        Args:
+            fn (str): Path to ICC Profile,
+            N (int): [1|3|4], # the numbers for colors 1=Gray, 3=RGB, 4=CMYK
+            alternate (str): ['DeviceGray'|'DeviceRGB'|'DeviceCMYK']
+        """
+        self.fn = fn
+        self.N = N
+        self.alternate = alternate
+
+
+class OutputIntentDictionary(ContentWithoutID):
+
+    __slots__ = (  # RAM usage optimization
+        "type",
+        "subtype",
+        "output_condition_identifier",
+        "output_condition",
+        "registry_name",
+        "dest_output_profile",
+        "info",
+    )
+
+    def __init__(
+        self,
+        subtype: OutputIntentSubType | str,
+        output_condition_identifier: str,
+        output_condition: str = None,
+        registry_name: str = None,
+        dest_output_profile: ICCProfileStreamDict | PDFICCProfileObject = None,
+        info: str = None,
+    ):
+        # super().__init__()
+        self.type = Name("OutputIntent")
+        self.subtype = Name(OutputIntentSubType.coerce(subtype).value)
```

Comment:
```suggestion
        "s",
        "output_condition_identifier",
        "output_condition",
        "registry_name",
        "dest_output_profile",
        "info",
    )

    def __init__(
        self,
        subtype: OutputIntentSubType | str,
        output_condition_identifier: str,
        output_condition: str = None,
        registry_name: str = None,
        dest_output_profile: ICCProfileStreamDict | PDFICCProfileObject = None,
        info: str = None,
    ):
        # super().__init__()
        self.type = Name("OutputIntent")
        self.s = Name(OutputIntentSubType.coerce(subtype).value)
```
This will be needed for the serialize() refactoring that I will suggest in the next comment.